File: src/dsa/Rotations.java
package dsa;

public class Rotations {
    //Using pointers
    public boolean areRotations(String str1, String str2) {

        if (str1.isEmpty() || str2.isEmpty() || str1 == null || str2 == null)
            throw new IllegalArgumentException();

        var pointer1 = 0;
        var pointer2 = str2.indexOf(str1.charAt(0));

        if (pointer2 == -1 || (str1.length() != str2.length()))
            return false;

        for (int i = 0; i < str1.length(); i++) {
            if (str1.charAt(pointer1) != str2.charAt(pointer2))
                return false;
            pointer1++;
            pointer2++;
            if (pointer2 == str2.length())
                pointer2 = 0;
        }
        return true;
    }
}
/*
ABCDE
CDEAB
*/

File: src/dsa-py/Rotations.py
def are_rotations(str1: str, str2: str):
    if len(str1) == 0 or len(str2) == 0 or str1 is None or str2 is None:
        raise ValueError("Invalid arguments")

    pointer1 = 0
    pointer2 = str2.find(str1[0])

    if pointer2 == -1 or len(str1) != len(str2):
        return False

    for i in range(len(str1)):
        if str1[pointer1] != str2[pointer2]:
            return False
        pointer1 += 1
        pointer2 += 1

        if pointer2 == len(str2):
            pointer2 = 0

    return True


result = are_rotations('ABCDE', 'CDEAB')
print(result)
